core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+    """
+    yaml_cont = {}
+    for line in file:
+        split_line = line.strip().split(":")
+        key = split_line[0].strip()
+        val = "".join(split_line[1:]).strip()
+        yaml_cont[key] = val
+    return yaml_cont
+
+
+def find_mutations_data(study_path):
+    """Find mutation data (MAF) file in cBioPortal study folder
+    
+    :param study_path: str, path to cBioPortal study folder
+    :return: str, name of mutations data (MAF) file
+    """
+    meta_files = glob.glob(study_path + "/meta*")
+    if len(meta_files) == 0:
+        raise FileNotFoundError("No meta files present in study")
+    for file in meta_files:
+        with open(file) as f:
+            yaml_cont = load_yaml(f)
+        if yaml_cont.get("stable_id") == "mutations" and yaml_cont.get("datatype") == "MAF":
+            # Correct file, retrieve associated datafile
+            print(f"Mutation meta file found in study: \'{file}\'")
+            return yaml_cont.get("data_filename"), yaml_cont.get("cancer_study_identifier")
+    raise FileNotFoundError(f"Mutation meta file not found in study \'{study_path}\'.")
+
+
+def acquire_maf_study_id(study_path):
+    """Find meta file in study and retrieve mutations file and study ID
+    
+    :param study_path: str, path to cBioPortal study
+    :return: str, mutation file path; str, study ID    
+    """
+    # Check if study exist
+    if not os.path.exists(study_path):
+        raise FileNotFoundError(f"{study_path} does not exist")
+    # Find meta files in study
+    mut_file, study_id = find_mutations_data(study_path)
+    mut_file = f"{study_path}/{mut_file}"
+    return mut_file, study_id
+
+
+def decipher_genome(maf_df):
+    """Try to get the genome build from a MAF file
+    
+    :param maf_df: pandas object, pandas dataframe containing 'NCBI_Build' column
+    :return: str, genome build (GRCh**)
+    """
+    genome = list(maf_df["NCBI_Build"].unique())
+    if len(genome) != 1:
+        raise ValueError(f"There are either no or multiple genome builds present in your MAF file: \'{genome}\'")
+    try:
+        genome = int(genome[0])
+    except ValueError:
+        try:
+            genome = int(genome[0][-2:])
+        except ValueError:
+            genome = input("Can't decipher genome build. Please enter manually (37/38): ")
+            genome = int(genome)
+    genome = "GRCh" + str(genome)
+    return genome
+
+
+def pre_process_maf(mut_file, tmp_dir):
+    """Pre-process the cBioPortal mutations data file for matrix generation 
+    
+    :param mut_file: str, path to mutations file
+    :param tmp_dir: str, path to temp folder
+    :return: path to pre-processed MAF, genome build
+    """
+
+    # Create temporary directory and copy file(s)
+    try:
+        os.mkdir(tmp_dir)
+    except FileExistsError:
+        answer = input(f"\nDirectory \'{tmp_dir}\' already exists. Remove directory (including contents) "
+                       f"and continue script? (y/n): ")
```

Comment:
I agree with Sander. We seldom have scripts that ask for an input once it's running, and imo that's better not to forget running processes. If really necessary, it can be added as another flag.